### ISSUE TITLE
Emulate CONNECT BY LEVEL < x and CONNECT BY ROWNUM < x

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11660]
Complete emulation of `CONNECT BY` is on the roadmap as https://github.com/jOOQ/jOOQ/issues/668, and is quite difficult to get right as a lot of syntactic elements interact with each other.

A much simpler usage of `CONNECT BY` is an undocumented and not officially supported one: The one that is similar to PostgreSQL's `GENERATE_SERIES`:

```sql
SELECT level
FROM dual
CONNECT BY level <= 10;
```

I.e. whenever there is no usage of `PRIOR` (as required by the docs), this behaviour is the same as:

```sql
SELECT level
FROM generate_series (1, 10) AS dual (level)
```

**Comment by lukaseder** [comment]
A test currently doesn't work on SQL Server because of a "transitive emulation" not working. I.e. this syntax is being rewritten to `GENERATE_SERIES` which is being rewritten to recursive CTE's, which are pushed up to the top level. So, if two subqueries use `CONNECT BY`, then we get wrong SQL:

```java
assertEquals(Arrays.asList(1, 2, 2, 3, 3, 4, 4, 4, 6, 6, 8, 8, 9, 12, 12, 16),
    create().select(field(name("t1", "c"), INTEGER).times(field(name("t2", "c"), INTEGER)))
            .from(
                select(level()).connectBy(level().lessThan(5)).asTable("t1", "c"),
                select(rownum()).connectBy(rownum().lessThan(5)).asTable("t2", "c"))
            .orderBy(1)
            .fetchInto(int.class));
```

This currently produces:

```sql
with 
  generate_series(generate_series) as (
    select 1
    union all
    select (generate_series + 1)
    from generate_series
    where generate_series < (5 - 1)
  ),
  generate_series(generate_series) as (
    select 1
    union all
    select (generate_series + 1)
    from generate_series
    where generate_series < (5 - 1)
  )

select ([t1].[c] * [t2].[c])
from
  (
    select level
    from (
      select generate_series
      from generate_series
    ) t (level)
  ) [t1] ([c]),
  (
    select rownum
    from (
      select generate_series
      from generate_series
    ) t (level)
  ) [t2] ([c])
order by 1
```

See also: https://github.com/jOOQ/jOOQ/issues/11777. We don't have to fix #11777 (difficult) in order to fix this. The name `generate_series` is irrelevant to user code, so we can generate a different alias.

**Comment by lukaseder** [comment]
The `ROWNUM` emulation also shows another problem:

```sql
    select rownum
    from (
      select generate_series
      from generate_series
    ) t (level)
```

The column is called `LEVEL` irrespective of how we're filtering. It should be called `ROWNUM`
